HACKING.md

```diff
@@ -1,162 +1,193 @@
+
+All PRs should pass unit tests. CI runs all tests automatically and won't merge a PR till they pass, so it is OK to create a PR and let CI run the e2e tests if you've done enough testing to be confident. This allows for human review and feedback while CI tests are running.
+
+To get a code review of unfinished work, create a PR with "[WIP]" at the start of the description. The CI system will not merge it until the "[WIP]" is removed and all tests are passing.
+
+## Setting up a test cluster
+
+Should work with any k8s cluster; tested with AWS and [Code Ready Containers](https://developers.redhat.com/products/codeready-containers/download). See below for extra steps with some specific cluster providers.
+
+Cluster creation requires a pull-secret from the cluster provider. You need
+to add a secret to this file so `make` can pull images from the openshift
+CI registry:
+
+Copy the `oc login` command from https://api.ci.openshift.org/oauth/token/request, then:
+```
+oc login api.ci.openshift.org ... # command from the web page
+oc registry login --to my-secret-file
+oc logout
+# This creates or adds to my-secret-file
+```
+
+Once created, you can `export KUBECONFIG=/path/to/my/cluster/config` or copy your config to `$HOME/.kube/config`.
+
+**Note**: Don't symlink `$HOME/.kube/config`. Some commands may overwrite this file and break the symlink, causing much confusion.
+
+Log in to your cluster as a user with the `cluster-admin` role. User `kubeadmin` is usually predefined with this role.
+
+### Configuration for CRC
+
+Download the crc command from [Code Ready Containers](https://developers.redhat.com/products/codeready-containers/download) and read the install instructions. 
+
+The following is suggested configuration to set before calling `crc start`
+
+```
+# Increase memory - default 8k may not be enough
+crc config set memory 12288
+# my-pull-secret is the CRC pull secret with the openshift CI secret added, explained above.
+crc config set pull-secret-file my-pull-secret
+# This test sometimes fails incorrectly on systems where libvirtd is triggered by a socket.
+crc config set skip-check-libvirt-running true
+```
+
+## Building and running tests
+
+Unit tests can be run without a cluster:
+* `make check`: Generate and format code, run unit tests and lint.
+   You should run *at least* this set of checks before a commit.
+
+To build, deploy and test the CLO image in your own cluster:
+* `make deploy`: Build CLO image and deploy to cluster with elasticsearch.
+* `make undeploy`: Undo `make deploy`.
+
+To run CLO as a local process:
+* `make run`: Run CLO as a local process. Does not require `make deploy`.
+* `make debug`: Run CLO in the `dlv` debugger.
+
+After running `make deploy`, `make run` or `make debug`, you can run e2e tests:
+* `make test-cluster` Run all e2e tests against the running CLO.
```

Comment:
i like `test-cluster` but it  doesn't seem that it obviously describes e2e or the path where you will find the tests

Reply:
Removed, this function is now done by test-e2e-local.